torch/accelerator/__init__.py

```diff
@@ -43,35 +46,47 @@ def is_available() -> bool:
 
         >>> assert torch.accelerator.is_available() "No available accelerators detected."
     """
-    return device_count() > 0
```

Comment:
I think, perhaps in some distribution scenarios, we expect `device_count` to not prevent feature fork as well. Should we delegate device_count to Python version of each backend and use `device_count() > 0` to check if the backend is available?